packages/wallet/Account.js

```diff
@@ -0,0 +1,550 @@
+const rp = require('request-promise');
+const StellarSdk = require('stellar-sdk');
+const Web3 = require('web3');
+const setting = require('./setting');
+const Trezor = require('./Trezor');
+const Ledger = require('./Ledger');
+
+const server = new StellarSdk.Server(setting.horizonEndpoint);
+const web3 = new Web3(setting.infuraUrl);
+
+/**
+ * @class Account
+ * This is a wrapper class over stellar and ethereum accounts.
+ * For those fields/methods that are already there when the original
+ * account is passed in, this class simply extracts them;
+ * Otherwise, the fields/methods are created and saved in this class.
+ */
+class Account {
+  /**
+  * constructor
+  * Setting balance to -1, to differentiate from accounts with 0 balance
+  * @param {string} network
+  */
+  constructor(network) {
+    this.balance = -1;
+    this.hardware = null;
+    this.nativeAccount = null;
+    this.history = null;
+    switch (network) {
+      case 'stellar':
+        this.network = 'stellar';
+        this.testAddress = `${setting.horizonEndpoint}/accounts/`;
+        StellarSdk.Network.useTestNetwork();
+        break;
+      case 'ethereum':
+        this.network = 'ethereum';
+        break;
+      default:
+        console.log(setting.networkError);
+    }
+  }
+
+  /**
+   * This is for stellar only
+   * @param {string} assetName
+   * @param {string} issuerPublickey
+   * @param {string} limit
+   * @returns {object|null} xdr.ChangeTrustOp if succeeds or null otherwise
+   */
+  async changeTrust(assetName, issuerPublickey, limit) {
+    switch (this.network) {
+      case 'stellar': {
+        const newAsset = new StellarSdk.Asset(assetName, issuerPublickey);
+        const receiver = await server.loadAccount(this.getAddress());
+        const transaction = new StellarSdk.TransactionBuilder(receiver)
+          .addOperation(StellarSdk.Operation.changeTrust({
+            asset: newAsset,
+            limit
+          }))
+          .build();
+        transaction.sign(this.nativeAccount);
+        return server.submitTransaction(transaction);
+      }
+      case 'ethereum':
+        console.log('Change trust is not for ethereum.');
+        break;
+      default:
+        console.log(setting.networkError);
+    }
+    return null;
+  }
+
+  /**
+  * set the balance field of this account
+  * @param {string} publicKey - the publick key of the account
+  */
+  async loadBalance(publicKey) {
+    try {
+      const response = await rp(this.testAddress + publicKey);
+      this.balance = JSON.parse(response).balances.slice(-1)[0].balance;// XLM balance
+    } catch (err) {
+      console.log('Error in loading balance');
+    }
+  }
+
+  /**
+   * Set the hardware of this account to be the hardware wallet object
+   * If using a hardware, this method must be called right after initialization.
+   * @param {string} hardware - Ledger of Trezor
+   */
+  setHardware(hardware) {
```

Comment:
Confusing concepts around `this.hardware`, `this.nativeAccount`, address and path.

I suggest we assign `Trezor` and `Ledger` object as `this.nativeAccount` but still keep a flag to indicate whether `Account` object is powered by hardware or not.

Even better, create a hardware account class  e.g `HardwareNative` where we can construct trezor, ledger and potentially future hardware wallets. E.g the construction of `HardwareNative` can be the assignment of `Trezor` object and it supports the methods of `HardwareNative`